examples/Training/java/src/training/ROIs.java

```diff
@@ -126,97 +117,114 @@ private void createROIs()
+
         //Polygon
-        Polygon polygon = new PolygonI();
-        polygon.setPoints(omero.rtypes.rstring(
-                "100.0,200.0 553.9,593.5 92.3,59.9"));
-        polygon.setTheZ(omero.rtypes.rint(0));
-        polygon.setTheT(omero.rtypes.rint(0));
-        roi.addShape(polygon);
+        List<Point2D.Double> points = new ArrayList();
+        points.add(new Point2D.Double(100.0,200));
+        points.add(new Point2D.Double(553.9,593.5));
+        points.add(new Point2D.Double(92.3,59.9));
+        PolygonData polygonData = new PolygonData(points);
+        polygonData.setZ(0);
+        polygonData.setT(0);
+        data.addShapeData(polygonData);
+
         //Polyline
-        Polyline polyline = new PolylineI();
-        polyline.setPoints(omero.rtypes.rstring(
-                "100.0,200.0 553.9,593.5 92.3,59.9"));
-        roi.addShape(polyline);
+        PolylineData polylineData = new PolylineData(points);
+        polylineData.setZ(0);
+        polylineData.setT(0);
+        data.addShapeData(polylineData);
+
         // Display fields which could quickly
         // be parsed from known formats
-        RInt GREY = omero.rtypes.rint(11184810);
-        Label text = new LabelI();
-        text.setTextValue(omero.rtypes.rstring("This is a polyline"));
-        text.setFontFamily(omero.rtypes.rstring("sans-serif"));
-        text.setFontSize(new LengthI(40, UnitsLength.POINT));
-        text.setFillColor(GREY);
-        text.setStrokeColor(GREY);
-        text.setStrokeWidth(new LengthI(25, UnitsFactory.Shape_StrokeWidth));
-        text.setLocked(omero.rtypes.rbool(true));
+        TextData textData = new TextData("This is a polyline", 10, 10);
+
+        Color fillcolor = new Color(128, 128, 128);
+        Color strokecolor = new Color(255, 255, 255);
+        textData.getShapeSettings().setFill(fillcolor);;
+        textData.getShapeSettings().setFontSize(new LengthI(40, UnitsLength.POINT));
+        textData.getShapeSettings().setFontFamily("sans-serif");
+        textData.getShapeSettings().setStroke(strokecolor);
+        textData.getShapeSettings().setStrokeWidth(new LengthI(25, UnitsFactory.Shape_StrokeWidth));
+        
         // Other options which may come with time
-        text.setFontStyle(omero.rtypes.rstring("italic"));
-        text.setFillColor(GREY);
-        text.setFillRule(omero.rtypes.rstring("even-odd"));
-        text.setStrokeColor(GREY);
-        text.setStrokeDashArray(omero.rtypes.rstring("10 20 30 10"));
-        text.setStrokeWidth(new LengthI(10, UnitsFactory.Shape_StrokeWidth));
-        text.setTheZ(omero.rtypes.rint(0));
-        text.setTheT(omero.rtypes.rint(0));
-        roi.addShape(text);
-        //Add a path shape
-        Path path = new PathI();
-        path.setD(omero.rtypes.rstring("M 100 100 L 300 100 L 200 300 z"));
-        roi.addShape(path);
+        textData.getShapeSettings().setFontStyle("italic");
+        textData.getShapeSettings().setFill(fillcolor);
```

Comment:
The logic was already there in the example, and I just replicated it, please take a look at this : https://github.com/openmicroscopy/openmicroscopy/blob/develop/examples/Training/java/src/training/ROIs.java#L189-L206

Guess the 2nd iteration of setting the shapesettings was to highlight the extra options that are available as well. Can remove the first bit, if this is confusing or misleading.